broker-daemon/models/block-order.js

```diff
@@ -159,8 +159,26 @@ class BlockOrder {
    * @return {BlockOrder} Modified block order instance
    */
   fail () {
+    // TODO: Do we need to fail the remaining orders that are tied to this block order in the ordersStore?
     this.status = BlockOrder.STATUSES.FAILED
+    return this
+  }
+
+  /**
+   * Move the block order to a completed status
+   * @return {BlockOrder} Modified block order instance
+   */
+  complete () {
+    const activeOrders = this.openOrders.filter(o => o.status === BlockOrder.STATUSES.ACTIVE)
+    const activeFills = this.fills.filter(f => f.status === BlockOrder.STATUSES.ACTIVE)
+
+    // If we still have active orders or fills, then the block order is not completed
+    // so we can simply return the object
+    if (activeOrders.length > 0 || activeFills.length > 0) {
+      return this
```

Comment:
shouldn't this be an error state of some kind?

Reply:
I made a comment above. We shouldn't complete an BlockOrder if there are existing active fills or orders. This is assuming that a BlockOrder -> has many orders/fills.